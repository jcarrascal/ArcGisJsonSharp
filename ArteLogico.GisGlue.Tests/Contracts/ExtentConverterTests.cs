using ApprovalTests;
using ArteLogico.GisGlue.Contracts;
using Newtonsoft.Json;
using NUnit.Framework;
using System.IO;

namespace ArteLogico.GisGlue.Tests.Contracts
{
    [TestFixture]
    public class ExtentConverterTests
    {
        [Test]
        public void WriteJson_WhenSerializingAnExtent_ThenTheFormatMatchesTheExample()
        {
            var extent = new Extent { X1 = 100, Y1 = 200, X2 = 1000, Y2 = 2000 };
            var serializer = JsonHelper.CreateSerializer();

            string result;
            using (var writer = new StringWriter())
            using (var jsonWriter = new JsonTextWriter(writer))
            {
                jsonWriter.Formatting = Formatting.Indented;
                serializer.Serialize(jsonWriter, extent);
                result = writer.ToString();
            }

            Approvals.Verify(result);
        }
    }
}