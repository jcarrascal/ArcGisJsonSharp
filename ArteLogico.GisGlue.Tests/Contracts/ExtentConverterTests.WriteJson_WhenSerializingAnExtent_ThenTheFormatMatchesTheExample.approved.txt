[
  [
    100.0,
    200.0
  ],
  [
    1000.0,
    2000.0
  ]
]